pkg/controller/pxc/controller.go

```diff
@@ -1095,3 +1058,77 @@ func createOrUpdateConfigmap(cl client.Client, configMap *corev1.ConfigMap) erro
 
 	return nil
 }
+
+func (r *ReconcilePerconaXtraDBCluster) createOrUpdate(obj runtime.Object) error {
+	metaAccessor, ok := obj.(metav1.ObjectMetaAccessor)
+	if !ok {
+		return errors.New("Can't convert object to ObjectMetaAccessor")
+	}
+
+	objectMeta := metaAccessor.GetObjectMeta()
+
+	if objectMeta.GetAnnotations() == nil {
+		objectMeta.SetAnnotations(make(map[string]string))
+	}
+
+	delete(objectMeta.GetAnnotations(), "percona.com/last_config_hash")
```

Comment:
Now this (and this `objectMeta.GetAnnotations()["percona.com/last_config_hash"] = hash`) works because `objectMeta.GetAnnotations()` return real annotation map of the object.
But  if in next version they decide to return copy of array it will break this logic.
So would suggest to treat with `objectMeta.GetAnnotations()` as it's already return copy of map.
Not sure if it worth the time to redo, but now it looks really unreliable.

Reply:
that's why we are using the `go.mod` file.
If we update the core library - we have to retest everything anyway. 
It could change, and could not. If it will change - we will do the required fixes

Reply:
Do we have tests for that?

Reply:
we will have them

Reply:
Let's wait for @dAdAbird opinion

Reply:
I agree with @blez, it looks a bit fishy. Despite GetAnnotations implementation would change or not it looks contradictional. Reading this code I want to check GetAnnotations implementation to understand that these assignments and deletes are legit.

I vote to change it for the sake of clarity:
```go
meta := objectMeta.GetAnnotations()
// do whatever you need
objectMeta.SetAnnotations(meta)
```

Reply:
a928f5e